spring-social-zotero/src/main/java/org/springframework/social/zotero/api/impl/GroupsTemplate.java

```diff
@@ -247,6 +249,55 @@ public void updateItem(String groupId, Item item, List<String> ignoreFields, Lis
             throw new ZoteroConnectionException("Could not update item.", e);
         }
     }
+    
+    
+    
+    /**
+     * This method makes a batch request call to Zotero to update items
+     * 
+     * @param groupId      group id of citations
+     * @param items        items that have to be updated
+     * @param ignoreFields fields that are not necessary while updating citations
+     * 
+     * @return ItemCreationResponse returns response from Zotero
+     */
+
+    /*
+     * The response returned by Zotero while creating a single item or updating
+     * multiple items is in the same format. So, we can use "ItemCreationResponse" data
+     * type to capture the response in both cases.
+     */
+    @Override
+    public ItemCreationResponse batchUpdateItems(String groupId, List<Item> items, List<String> ignoreFields)
+            throws ZoteroConnectionException {
+        if (items.size() > ZOTERO_BATCH_UPDATE_LIMIT) {
+            throw new IllegalArgumentException(
+                    String.format("Items size cannot be more than %s", ZOTERO_BATCH_UPDATE_LIMIT));
+        }
```

Comment:
let's have this plugin send multiple requests if there are more items than the limit passed. Otherwise, everywhere we use this method, the same code has to be repeated to make multiple requests.

Reply:
If we go with this approach there is a possibility that request times out from front-end call. Say, if we are getting some 500 items from the front-end by the time we do multiple request and send back the response, the call from front-end could have already timed out.